File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        charSet = set()
        result = 1

        for i in range(len(s)):
            if s[i] in charSet:
                result += 1
                charSet = set()
            charSet.add(s[i])

        return result

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    int partitionString(string s) {
        set<char> charSet;
        int result = 1;
        for (const char &c : s) {
            if (charSet.contains(c)) {
                ++result;
                charSet.clear();
            }
            charSet.insert(c);
        }
        return result;
    }
};